### PR TITLE
PMM-9237-re-enabling-tests-migrate-from-enzyme: fix unit tests in add…

### PR DESCRIPTION
**Pull request by solovevayaroslavna**
**What this PR does / why we need it**:

Restoring previously disabled tests, reducing the number of potential errors in grafanaFolder

**Which issue(s) this PR fixes**:
https://jira.percona.com/browse/PMM-9237

### CODE REVIEW

File: public/app/percona/add-instance/components/AddRemoteInstance/FormParts/HAProxyConnectionDetails/HAProxyConnectionDetails.test.tsx [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
+describe('HAProxy connection details:: ', () => {
   it('should trim username and password values right', async () => {
-    const root = mount(
-      <Form onSubmit={jest.fn()} render={() => <HAProxyConnectionDetails remoteInstanceCredentials={{}} />} />
-    );
+    render(<Form onSubmit={jest.fn()} render={() => <HAProxyConnectionDetails remoteInstanceCredentials={{}} />} />);
 
-    root.find(dataTestId('username-text-input')).simulate('change', { target: { value: '    test    ' } });
-    root.find(dataTestId('password-password-input')).simulate('change', { target: { value: '    test    ' } });
+    const userNameTextInput = await screen.getByTestId('username-text-input');
```
Comment (tiagomotasantos) [inline public/app/percona/add-instance/components/AddRemoteInstance/FormParts/HAProxyConnectionDetails/HAProxyConnectionDetails.test.tsx:10]: No need to await when using get as it doesn't return a promise, also this test doesn't need to be async anymore

File: public/app/percona/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocationModal.test.tsx [hunk 2]
```diff
@@ -35,28 +28,32 @@ xdescribe('AddStorageLocationModal', () => {
...
+    render(<AddStorageLocationModal location={location} onClose={jest.fn()} onAdd={jest.fn()} isVisible />);
+    expect(screen.queryByTestId('bucketName-field-container')).toBeInTheDocument();
+    expect(screen.queryByTestId('accessKey-field-container')).toBeInTheDocument();
+    expect(screen.queryByTestId('secretKey-field-container')).toBeInTheDocument();
+    expect(screen.queryByTestId('endpoint-field-container')).toBeInTheDocument();
+    expect(screen.queryByRole('textbox', { name: /endpoint/i })).toHaveValue('/foo/bar');
   });
 
-  it('should not render unknown type fields', () => {
+  it('should not render unknown type fields', async () => {
```
Comment (tiagomotasantos) [inline public/app/percona/backup/components/StorageLocations/AddStorageLocationModal/AddStorageLocationModal.test.tsx:39]: Don't need to be async as we are not using await inside the test

File: public/app/percona/dbaas/components/DBCluster/AddDBClusterModal/AddDBClusterModal.test.tsx [hunk 3]
```diff
@@ -1,28 +1,26 @@
...
-    root
-      .find(`[data-testid="${step}"]`)
-      .find('[data-testid="step-content"]')
-      .find('div')
-      .at(1)
-      .prop('className')
-      ?.includes('current');
+  const isStepActive = (step: string) => {
+    const stepNode = screen.getByTestId(`${step}`).querySelector('[data-testid="step-content"]');
+    return stepNode ? stepNode.getElementsByTagName('div')[0].className.split('-')?.includes('current') : undefined;
```
Comment (tiagomotasantos) [inline public/app/percona/dbaas/components/DBCluster/AddDBClusterModal/AddDBClusterModal.test.tsx:19]:
```suggestion
    return stepNode ? stepNode.getElementsByTagName('div')[0].className.includes('current') : false;
```

File: public/app/percona/dbaas/components/StepProgress/StepProgress.test.tsx [hunk 4]
```diff
@@ -27,26 +27,26 @@ xdescribe('StepProgress::', () => {
...
-      .find(`[data-testid="${dataTestId}"]`)
-      .find('[data-testid="step-content"]')
-      .find('div')
-      .at(1)
-      .prop('className')
-      ?.includes('current');
+  const isCurrentStep = (dataTestId: string) => {
+    const stepContent = screen.getByTestId(`${dataTestId}`).querySelector('[data-testid="step-content"]');
+    return stepContent
+      ? stepContent.getElementsByTagName('div')[0].className.split('-')?.includes('current')
```
Comment (tiagomotasantos) [inline public/app/percona/dbaas/components/StepProgress/StepProgress.test.tsx:33]: Same as above

**Review by tiagomotasantos — CHANGES_REQUESTED** [review]
Mainly some problems with async/await, other than that it's very good! Please recheck if async can be removed from other tests definition

File: public/app/percona/add-instance/components/AddRemoteInstance/FormParts/HAProxyConnectionDetails/HAProxyConnectionDetails.test.tsx (hunk 1 above)
Comment (solovevayaroslavna) [reply public/app/percona/add-instance/components/AddRemoteInstance/FormParts/HAProxyConnectionDetails/HAProxyConnectionDetails.test.tsx:10]: Yes, I think that at this point you are definitely right, I will review the tests again. Probably here I forgot to delete async after fixing find on get.

**Comment by atymchuk** [comment]
It's a great contribution, thanks a lot!

**Review by atymchuk — DISMISSED** [review]

File: public/app/percona/dbaas/components/DBCluster/AddDBClusterModal/AddDBClusterModal.test.tsx (hunk 3 above)
Comment (solovevayaroslavna) [reply public/app/percona/dbaas/components/DBCluster/AddDBClusterModal/AddDBClusterModal.test.tsx]: fixed

File: public/app/percona/dbaas/components/StepProgress/StepProgress.test.tsx (hunk 4 above)
Comment (solovevayaroslavna) [reply public/app/percona/dbaas/components/StepProgress/StepProgress.test.tsx]: fixed

**Review by tiagomotasantos — APPROVED** [review]

**Review by fabio-silva — APPROVED** [review]

**Pull request closed without merging**